src/Yarhl/IO/StreamFormat/StreamWrapper.cs

```diff
@@ -51,6 +53,14 @@ protected StreamWrapper()
             protected set;
         }
 
+        /// <summary>
+        /// Gets or sets the lock object of this stream (and all its substreams).
+        /// </summary>
+        public object LockObj {
+            get;
+            set;
```

Comment:
Can we just implement the `get` (no the `set`). If there isn't any use case I think it's safer to just provide the internal lock instead of allowing setting it from outside.